packages/dataviews/src/field-types/number.tsx

```diff
@@ -0,0 +1,118 @@
+/**
```

Comment:
I think it'd be better to create a `utils/validated-number.tsx` utility (see existing `utils/validated-input.tsx`) that it's used by both `integer` and `decimal` controls.

Note these controls are used in both Edit and the filters, so we need to test everything there work properly.